riscof-plugin/sail_cSim/riscof_sail_cSim.py

```diff
@@ -0,0 +1,213 @@
+import riscof.constants as constants
+from riscv_isac.isac import isac
+
+logger = logging.getLogger()
+
+class sail_cSim(pluginTemplate):
+    __model__ = "sail_c_simulator"
+    __version__ = "0.5.0"
+
+    def __init__(self, *args, **kwargs):
+        sclass = super().__init__(*args, **kwargs)
+
+        config = kwargs.get('config')
+	# Check if the config.ini node has a sail_cSim node. Else error out.
+        if config is None:
+            logger.error("Config node for sail_cSim missing.")
+            raise SystemExit
+
+        # capture the number of parallel jobs that can be used for running tests
+        # on sail_cSim either from the config.ini or default to 1
+        self.num_jobs = str(config['jobs'] if 'jobs' in config else 1)
+
+	# capture tha path of the plugin from the config.ini node.
+        self.pluginpath = os.path.abspath(config['pluginpath'])
+
+	# since sail_csim has different binaries for 32 and 64. Here we have
+	# created a small dictionary which maps XLEN to the respective binary paths. 
+	# We first check for the PATH variable in the config.ini, else we default to 
+	# riscv_sim_RV32/RV64. The element selection happens in the build phase, when we 
+	# access to the ISA yaml (coming from DUT).
+        self.sail_exe = { '32' : os.path.join(config['PATH'] if 'PATH' in config else "","riscv_sim_RV32"),
+                '64' : os.path.join(config['PATH'] if 'PATH' in config else "","riscv_sim_RV64")}
+	
+	# This is usually not required unless you are running sail_cSim as a DUT.
+	# This variable is used to set path of the riscv-config isa spec if specified in the
+	# config.ini file. Else default to an empty string.
+        self.isa_spec = os.path.abspath(config['ispec']) if 'ispec' in config else ''
+
+	# This is usually not required unless you are running sail_cSim as a DUT.
+	# This variable is used to set path of the riscv-config platform spec if specified in the
+	# config.ini file. Else default to an empty string.
+        self.platform_spec = os.path.abspath(config['pspec']) if 'ispec' in config else ''
+
+	# capture the make command from the config.ini which could
+        # default to using "make". Options for this command could be "pmake" as
+        # well.
+        self.make = config['make'] if 'make' in config else 'make'
+
+        # logger statements to print on the terminal during the run
+        logger.debug("SAIL CSim plugin initialised using the following configuration.")
+        for entry in config:
+            logger.debug(entry+' : '+config[entry])
+        return sclass
+
+    def initialise(self, suite, work_dir, archtest_env):
+
+	# capture the suite directory
+        self.suite = suite
+	
+	# capture the working directory.
+        self.work_dir = work_dir
+
+	# set the compile and objdump commands here. Note the march is not hardwired here, 
+	# because it will change for each test.
+        self.objdump_cmd = 'riscv{1}-unknown-elf-objdump -D {0} > {2};'
```

Comment:
You need to shell escape everything (and does your library's add_target correctly escape that again for make?)